auth/env_keychain.go

```diff
@@ -23,9 +27,18 @@ func DefaultKeychain(images ...string) (authn.Keychain, error) {
 		return nil, err
 	}
 
+	// note: context is not used in the NewNoClient path
+	// this adds a credential provider-like keychain for public cloud providers
+	klog.SetLogger(logr.Discard())
```

Comment:
There didn't seem to be a way to go about discarding logs in any other way. `k8schain` uses `github.com/vdemeester/k8s-pkg-credentialprovider/aws ` which uses `klog` and doesn't seem to be any inputs to send in a customized logger per invocation.